synth-converter/src/main.rs

```diff
@@ -1,3 +1,63 @@
-fn main() {
-    println!("Hello, world!");
+use anyhow::{Context, Result};
+use clap::Parser;
+use std::{
+    fs::File,
+    io::{Read, Write},
+    path::Path,
+};
+use synth_converter::convert::json_to_rdf;
+
+/// Converts CAT+ Synthesis JSON input into RDF formats.
+///
+/// This tool expects Synthesis data similar to example/1-Synth.json
+/// of a batch with actions. This data is then transformed to RDF and
+/// serialized as Turtle (ttl) or JSON-LD (jsonld).
+#[derive(Parser, Debug)]
+struct Args {
+    /// Path to the input JSON file: relative or absolute
+    input_file: String,
+
+    /// Path to the output RDF file
+    output_file: String,
+
+    /// Output format: "ttl" (Turtle) or "jsonld" (JSON-LD)
+    #[arg(short, long, default_value = "ttl")]
+    format: String,
+}
+
+fn main() -> Result<()> {
+    let args = Args::parse();
+
+    // Validate input file
+    let input_path = Path::new(&args.input_file);
+    if !input_path.exists() {
+        anyhow::bail!("Input file '{}' does not exist.", args.input_file);
+    }
+    if !input_path.is_file() {
+        anyhow::bail!("'{}' is not a valid file.", args.input_file);
+    }
```

Comment:
suggestion: instead of using a comment just make a function `validateInput` or so...